frontend/src/hooks/useChess.js

```diff
@@ -400,7 +400,6 @@ export const useChess = ({ onPlayerMove, onBotMove, lock, isPaused, config, setC
     setMoveTo(null);
     setShowPromotionDialog(false);
     setOptionSquares({});
-    if (gameMode === 'timed') handleMoveTimerSwitch();
```

Comment:
Does the timer switch on promotion after this is removed? I was rather thinking of moving it to the end of the `if (piece)` check, with the additional check of `move` existence, that is in `useChess`

```js
  const onPromotionPieceSelect = (piece, fromSquare, toSquare) => {
    .
    .
    .

      if (move && onPlayerMove) onPlayerMove(move, prevFen, game.fen());
      if (move && gameMode === 'timed') handleMoveTimerSwitch();
      resetArrows();
    }
...
```